src/lib.rs

```diff
@@ -1,38 +1,45 @@
-extern crate serde;
-#[macro_use]
-extern crate serde_derive;
-extern crate toml;
-
-use serde::de;
-use serde::de::{value, Deserialize, Deserializer, SeqAccess, Visitor};
-use serde::ser::{Serialize, SerializeSeq, Serializer};
-use std::collections::{HashMap, HashSet};
-use std::fmt;
+use serde::{
+    de::{self, value, Deserializer, SeqAccess, Visitor},
+    ser::{SerializeSeq, Serializer},
+    Deserialize, Serialize,
+};
+use std::{
+    collections::{HashMap, HashSet},
+    fmt,
+};
+use toml::de::Error;
 
 /// Config represents the full configuration within a netlify.toml file.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Default)]
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
 pub struct Config {
     pub build: Option<Context>,
     pub context: Option<HashMap<String, Context>>,
     pub redirects: Option<Vec<Redirect>>,
     pub headers: Option<Vec<Header>>,
     pub template: Option<Template>,
+    #[serde(
+        alias = "edgeHandlers",
+        alias = "edge-handlers",
+        alias = "edge_handlers",
+        default
+    )]
+    pub edge_handlers: Vec<EdgeHandler>,
 }
 
 /// Context holds the build variables Netlify uses to build a site before deploying it.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Default)]
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
 pub struct Context {
     pub base: Option<String>,
     pub publish: Option<String>,
     pub command: Option<String>,
     pub functions: Option<String>,
     pub environment: Option<HashMap<String, String>>,
-    #[serde(alias = "edge-handlers")]
+    #[serde(alias = "edge-handlers", alias = "edgeHandlers")]
     pub edge_handlers: Option<String>,
 }
 
 /// Redirect holds information about a url redirect.
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
```

Comment:
as a follow-up: if we want to warn people about the edge-handler prop being ignored we could use https://docs.rs/serde_ignored/0.1.2/serde_ignored/